File: cpp/bc2152.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main()
{
	int ho, mi, val, count;
	string mes = "";
	
	cin >> count;
	
	for(int i = 0; i < count;i++)
	{
		cin >> ho >> mi >> val;
		
		mes = val == 1? "A porta abriu!" : "A porta fechou!";
		
		cout << setw(2) << setfill('0') << ho << ":" << setw(2) << setfill('0')<< mi << " - " << mes << endl;
		
	}
}

File: python/bc2152.py
count = int(input())

for c in range(0, count):
    cVal = input().split(" ")

    ho = int(cVal[0])
    mi = int(cVal[1])
    val = int(cVal[2])

    res = "A porta abriu!" if val == 1 else "A porta fechou!"

    print(f"{ho:02d}:{mi:02d} - {res}")
